### PR TITLE
Allow running android smoketests in CI [ci smoketest fenix=master a-c]

### PR DESCRIPTION
**Pull request by rfk**
A little experiment to see if I can do an android version of https://github.com/mozilla/application-services/pull/2833, using taskcluster instead of circle. May be risky given the refactoring that's happening to our taskcluster setup, but we'll see...

---

This adds support for triggering fenix and/or a-c smoketests directly
from a pull-request, by including special syntax in the PR title.
Examples:

  * `[ci smoketest fenix]`: run fenix smoketests using default branch
  * `[ci smoketest fenix a-c]`: run both fenix and a-c smoketests
  * `[ci smoketest fenix=foobar]`: run fenix smoktests using branch `foobar`
  * `[ci smoketest fenux]`: dont run any tests; typos are ignored

### CODE REVIEW

**Comment by rfk** [comment]
To my light surprise, this actually succeeded at running the smoketests, including re-triggering them if I edit the PR title.

They don't actually *pass* in CI, because of a build failure when generating the NSS bindings:

```
error: header '/build/repo/libs/android/armeabi-v7a/nss/include/nss/secoidt.h' does not exist.
```

@eoger have you seen this before?

I'm going to see if I can limit it to not build for that platform and whether that makes a difference...

**Comment by rfk** [comment]
> I'm going to see if I can limit it to not build for that platform and whether that makes a difference...

 This worked; calling this one ready for r?

File: automation/taskcluster/decision_task.py [hunk 1]
```diff
@@ -27,6 +49,13 @@ def main(task_for):
             android_multiarch()
         else:
             android_linux_x86_64()
+        smoketests = SMOKETEST_CI_TAG_RE.search(pr_title)
+        if smoketests is not None:
+            if smoketests.group("fenix") is not None:
+                fenix_smoketest(smoketests.group("fenix_branch"),
+                                smoketests.group("a_c_branch"))
+            if smoketests.group("a_c") is not None:
+                android_components_smoketest(smoketests.group("a_c_branch"))
```
Comment (eoger) [inline automation/taskcluster/decision_task.py:58]: IIUC, `[ci smoketest fenix=foo a-c=bar]` would result in 2 tasks being created:
- a-c smoketests with branch `bar`
- `fenix smoketests with branch `foo` against a-c branch `bar`

File: automation/shared.py [hunk 2]
```diff
@@ -60,5 +61,6 @@ def set_gradle_substitution_path(project_dir, name, value):
         relpath /= ".."
     for nm in abs_value.parts[len(ancestor.parts):]:
         relpath /= nm
+    step_msg(f"Setting relative path from {project_dir} to {abs_value} as {relpath}")
     with properties_file.open("a") as f:
-        f.write(f"{name}={abs_value}\n")
\ No newline at end of file
+        f.write(f"{name}={relpath}\n")
```
Comment (eoger) [inline automation/shared.py:66]: Hit that problem today, thanks for the fix!

File: automation/taskcluster/decision_task.py (hunk 1 above)
Comment (rfk) [reply automation/taskcluster/decision_task.py:58]: Correct.

**Pull request merged**